webhooks/webhooks_test.go

```diff
@@ -557,6 +748,190 @@ func TestParse_SuccessfulRefundNotification(t *testing.T) {
+	xmlFile := MustOpenFile("testdata/processing_payment_notification.xml")
+	if result, err := webhooks.Parse(xmlFile); err != nil {
+		t.Fatal(err)
+	} else if n, ok := result.(*webhooks.PaymentNotification); !ok {
+		t.Fatalf("unexpected type: %T, result", n)
+	} else if diff := cmp.Diff(n, &webhooks.PaymentNotification{
+		Type: webhooks.ProcessingPayment,
+		Account: webhooks.Account{
+			XMLName:     xml.Name{Local: "account"},
+			Code:        "1",
+			Username:    "verena",
+			Email:       "verena@example.com",
+			FirstName:   "Verena",
+			LastName:    "Example",
+			CompanyName: "Company, Inc.",
+		},
+		Transaction: webhooks.Transaction{
+			XMLName:          xml.Name{Local: "transaction"},
+			UUID:             "a5143c1d3a6f4a8287d0e2cc1d4c0427",
+			InvoiceNumber:    2059,
+			SubscriptionUUID: "1974a098jhlkjasdfljkha898326881c",
+			Action:           "purchase",
+			AmountInCents:    1000,
+			Status:           "processing",
+			Message:          "Bogus Gateway: Forced success",
+			Reference:        "",
+			Source:           "subscription",
+			Test:             recurly.NullBool{Valid: true, Bool: true},
+			Voidable:         recurly.NullBool{Valid: true, Bool: true},
+			Refundable:       recurly.NullBool{Valid: true, Bool: true},
+		},
+	}); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestParse_ScheduledPaymentNotification(t *testing.T) {
+	xmlFile := MustOpenFile("testdata/scheduled_payment_notification.xml")
+	if result, err := webhooks.Parse(xmlFile); err != nil {
+		t.Fatal(err)
+	} else if n, ok := result.(*webhooks.PaymentNotification); !ok {
+		t.Fatalf("unexpected type: %T, result", n)
+	} else if diff := cmp.Diff(n, &webhooks.PaymentNotification{
+		Type: webhooks.ScheduledPayment,
+		Account: webhooks.Account{
+			XMLName:     xml.Name{Local: "account"},
+			Code:        "1",
+			Username:    "verena",
+			Email:       "verena@example.com",
+			FirstName:   "Verena",
+			LastName:    "Example",
+			CompanyName: "Company, Inc.",
+		},
+		Transaction: webhooks.Transaction{
+			XMLName:          xml.Name{Local: "transaction"},
+			UUID:             "a5143c1d3a6f4a8287d0e2cc1d4c0427",
+			InvoiceNumber:    2059,
+			SubscriptionUUID: "1974a098jhlkjasdfljkha898326881c",
+			Action:           "purchase",
+			AmountInCents:    1000,
+			Status:           "scheduled",
+			Message:          "Bogus Gateway: Forced success",
+			Reference:        "",
+			Source:           "subscription",
+			Test:             recurly.NullBool{Valid: true, Bool: true},
+			Voidable:         recurly.NullBool{Valid: true, Bool: true},
+			Refundable:       recurly.NullBool{Valid: true, Bool: true},
+		},
+	}); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestParse_NewDunningEventNotification(t *testing.T) {
+	fmt.Println("Available after credit invoices have been turned on in your account. See https://docs.recurly.com/docs/credit-invoices-release#section-updates-to-the-api-and-sdks for more information.")
```

Comment:
Let's delete this line and the blank line below it. There's an announcement in the `README` of this repo explaining this.